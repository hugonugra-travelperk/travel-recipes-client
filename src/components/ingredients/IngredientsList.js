import { v4 as uuidv4 } from 'uuid';
import { CardSubBody } from "../base/Card";
import {ListItem } from "../base/List"
import IngredientElement from './IngredientElement'

export default function IngredientsList(props){
  return(
    <CardSubBody>
      <ul>
        {props.ingredients.map(function(item){
          return (
            <ListItem key={uuidv4()}>
              <IngredientElement ingredient={item}/>
            </ListItem>
          )
        })}
      </ul>
    </CardSubBody>
  )
}
