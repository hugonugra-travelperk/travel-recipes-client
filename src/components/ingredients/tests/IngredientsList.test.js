import {render, cleanup} from "@testing-library/react"
import "@testing-library/jest-dom";
import IngredientsList from "../IngredientsList"

afterEach(cleanup);

it("renders", () => {
  const {asFragment} = render(<IngredientsList ingredients={[{'id':'1', 'name': 'test text'}]}/>);
  expect(asFragment()).toMatchSnapshot();
});
