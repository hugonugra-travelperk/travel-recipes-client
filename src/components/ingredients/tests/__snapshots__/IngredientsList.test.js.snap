// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="sc-eCImPb gayBGf"
  >
    <ul>
      <li
        class="sc-jRQBWg kEHRpN"
      >
        <span
          class="sc-gKclnd CJqjG"
        >
          test text
        </span>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
